tests/test_watcher_hooks.py

```diff
@@ -0,0 +1,35 @@
+from observ import reactive, watch
+from observ.watcher import Watcher
+
+
+def test_watcher_hooks():
+    # demonstrate that watchers can be kept alive in a global registry
+    # using on_created and on_destroyed hooks
+    watchers = []
+
+    def on_created(watcher):
+        watchers.append(watcher)
+
+    def on_destroyed(watcher):
+        watchers.remove(watcher)
+
+    try:
+        Watcher.on_created = on_created
+        Watcher.on_destroyed = on_destroyed
+
+        a = reactive([1, 2])
+        called = 0
+
+        def _callback():
+            nonlocal called
+            called += 1
+
+        watch(lambda: len(a), _callback, sync=True)
+        assert called == 0
+        a.append(3)
+        assert called == 1
+        assert len(a) == 3
```

Comment:
I'm missing a check for the length of the watchers array here. In its current state it is not testing that much :)

Reply:
Well without the watchers array, it won't be called!

Reply:
I extended the test anyway :)